packages/mui-material/src/Dialog/Dialog.d.ts

```diff
@@ -52,6 +52,10 @@ export interface DialogProps extends StandardProps<ModalProps, 'children'> {
    * @deprecated Use the `onClose` prop with the `reason` argument to handle the `backdropClick` events.
    */
   onBackdropClick?: ModalProps['onBackdropClick'];
+  /**
+   * @ignore
+   */
+  onClick?: ModalProps['onClick'];
```

Comment:
We don't need to include it in the type definition. Since `DialogProps` extends `ModalProps`, `DialogProps` will inherit the `onClick` property from `ModalProps`.
```suggestion
```